### PR TITLE
Fix update scenario for network interface

### PR DESCRIPTION
**Pull request by neil-yechenwei**
fixes terraform-providers/terraform-provider-azurerm#6546

The update API of network interface is full update not incremental update.

### CODE REVIEW

File: azurerm/internal/services/network/tests/resource_arm_network_interface_test.go [hunk 1]
```diff
@@ -444,6 +469,66 @@ resource "azurerm_network_interface" "test" {
...
+  name                    = "acctestni-%d"
+  location                = azurerm_resource_group.test.location
+  resource_group_name     = azurerm_resource_group.test.name
+  enable_ip_forwarding    = true
+  internal_dns_name_label = "acctestni-%s"
+
+  dns_servers = [
+    "10.0.0.5",
+    "10.0.0.6"
+  ]
```
Comment (jackofallops) [inline azurerm/internal/services/network/tests/resource_arm_network_interface_test.go:517]: The linked issue refers to changing DNS servers, but these stay the same through the update test, should we be testing/validating this in the update?

Comment (neil-yechenwei) [reply azurerm/internal/services/network/tests/resource_arm_network_interface_test.go]: updated

**Comment by neil-yechenwei** [comment]
@jackofallops , thanks for your comments. I've updated code.

**Comment by jackofallops** [comment]
Tests pass: 
image

**Pull request merged**